src/components/Card.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { SyntheticEvent, useRef, useState } from "react";
+import styles from "./Card.module.css";
+import { TextInput } from "./TextInput";
+import { SmallIconButton } from "./SmallIconButton";
+import { TextOutput } from "./TextOutput";
+import { BigButton } from "./BigButton";
+
+interface CardProps {
+  /**text on side A */
+  faceValue?: string;
+  /**text on side B */
+  flipValue?: string;
+}
+
+export const Card = (props: CardProps) => {
+  // text on side A
+  const [faceValue, setFaceValue] = useState("");
+  const [tempFaceValue, setTempFaceValue] = useState(faceValue);
+  // text on side B
+  const [flipValue, setFlipValue] = useState("");
+
+  //wether we're editing the card
+  const [editEnabled, setEditEnabled] = useState(false);
+
+  //flipState on = side A displayed, flipState off = side B
+  const [flipState, setFlipState] = useState(true);
+
+  //state used for controlling the default display of input
+  const [inputDisplayValue, setInputDisplayValue] = useState(
+    flipState ? faceValue : flipValue
```

Comment:
**issue:** This will work only on initial render. To have it done properly you would have to use `useEffect`. Eg:

```jsx
useEffect(() => {
   const displayValue = flipState ? faceValue : flipValue
   setInputDisplayValue(displayValue)
}, [flipState])
```

Reply:
Actually it did work, because of lines 39, 50 & 60 in the same file. But I applied `useEffect` anyway, and could remove those three lines.